apps/comps/rpc/router/user/get-profile.ts

```diff
@@ -0,0 +1,56 @@
+import { SelectUser } from "@recallnet/db/schema/core/types";
+import { ActorStatus } from "@recallnet/services/types";
+
+import { base } from "@/rpc/context/base";
+import { authMiddleware } from "@/rpc/middleware/auth";
+import { User as FrontendUser } from "@/types";
+
+/**
+ * Serialize database user to frontend format
+ * Converts null to undefined and Date objects to ISO strings
+ */
+function serializeUser(dbUser: SelectUser): FrontendUser {
+  return {
+    id: dbUser.id,
+    walletAddress: dbUser.walletAddress,
+    embeddedWalletAddress: dbUser.embeddedWalletAddress ?? undefined,
+    walletLastVerifiedAt: dbUser.walletLastVerifiedAt?.toISOString(),
+    name: dbUser.name ?? undefined,
+    email: dbUser.email ?? "",
+    isSubscribed: dbUser.isSubscribed,
+    privyId: dbUser.privyId ?? undefined,
+    imageUrl: dbUser.imageUrl ?? undefined,
+    metadata: dbUser.metadata
+      ? (dbUser.metadata as { website?: string })
+      : undefined,
+    status: dbUser.status as ActorStatus,
+    createdAt: dbUser.createdAt.toISOString(),
+    updatedAt: dbUser.updatedAt.toISOString(),
+    lastLoginAt: dbUser.lastLoginAt?.toISOString(),
+  };
+}
+
+export const getProfile = base
+  .use(authMiddleware)
+  .handler(async ({ context, errors }) => {
+    try {
+      const user = await context.userService.getUser(context.user.id);
+
+      if (!user) {
+        throw errors.NOT_FOUND();
+      }
```

Comment:
Ok, so cool thing is the user you are querying already exists at `context.user`, so you don't have to query for it again. Just return it. And it will always be found because you used the auth middleware on this procedure, so for the handler to even run implies the user must exist.

Reply:
oh yeah.  I totally knew that and did this the right way in an earlier version of this PR, but then forgot about it this time around 🤦.  Good call